File: Data Structures & Algorithms/balanced-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {boolean}
     */
    isBalanced(root) {
        let res = 0
        let flag = true
        
        const depth = (node) => {
            if(node === null)
                return 0
            
            // tính chiều sâu hai nhánh
            let left = depth(node.left)
            let right = depth(node.right)

            //Cập nhật chênh lệch độ sâu
            res = Math.max(res, Math.abs(left-right))

            //return chiều sau của node
            return 1+Math.max(left,right)
        }

        depth(root)
        return res <= 1 ? true : false
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public List<int> RightSideView(TreeNode root) {
        var res = new List<int>();
        DFS(root, 0, res);

        return res;
    }

    public void DFS(TreeNode node, int level, List<int> res)
    {
        if(node == null)
            return;
        
        if(level == res.Count)
            res.Add(node.val);
        
        DFS(node.right, level + 1, res);
        DFS(node.left, level + 1, res);
    }
}
